From cdb33c3d5621ce035ad6950c8e2268fe94b73de5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Tue, 10 Jun 2025 06:46:30 -0700
Subject: [PATCH] macOS: Remove linkage to AGL framework

It's no longer available on macOS 26, and we don't use it anymore
anyways.

Pick-to: 6.10 6.9 6.8 6.5
Change-Id: Ia1d0e37dda177f333646e598e517f4af20215dad
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
---
 cmake/FindWrapOpenGL.cmake | 9 ---------
 mkspecs/common/mac.conf    | 5 ++---
 2 files changed, 2 insertions(+), 12 deletions(-)

diff --git a/qtbase/cmake/FindWrapOpenGL.cmake b/qtbase/cmake/FindWrapOpenGL.cmake
index 7295a159caf6..fe73ab782118 100644
--- a/qtbase/cmake/FindWrapOpenGL.cmake
+++ b/qtbase/cmake/FindWrapOpenGL.cmake
@@ -37,16 +37,7 @@ if (OpenGL_FOUND)
             set(__opengl_fw_path "-framework OpenGL")
         endif()

-        find_library(WrapOpenGL_AGL NAMES AGL)
-        if(WrapOpenGL_AGL)
-            set(__opengl_agl_fw_path "${WrapOpenGL_AGL}")
-        endif()
-        if(NOT __opengl_agl_fw_path)
-            set(__opengl_agl_fw_path "-framework AGL")
-        endif()
-
         target_link_libraries(WrapOpenGL::WrapOpenGL INTERFACE ${__opengl_fw_path})
-        target_link_libraries(WrapOpenGL::WrapOpenGL INTERFACE ${__opengl_agl_fw_path})
     else()
         target_link_libraries(WrapOpenGL::WrapOpenGL INTERFACE OpenGL::GL)
     endif()
diff --git a/mkspecs/common/mac.conf b/mkspecs/common/mac.conf
index 61bea952b220..9ba38d9949d6 100644
--- a/qtbase/mkspecs/common/mac.conf
+++ b/qtbase/mkspecs/common/mac.conf
@@ -18,8 +18,7 @@ QMAKE_LIBDIR            =

 # sdk.prf will prefix the proper SDK sysroot
 QMAKE_INCDIR_OPENGL     = \
-    /System/Library/Frameworks/OpenGL.framework/Headers \
-    /System/Library/Frameworks/AGL.framework/Headers/
+    /System/Library/Frameworks/OpenGL.framework/Headers

 QMAKE_FIX_RPATH         = install_name_tool -id

@@ -30,7 +29,7 @@ QMAKE_LFLAGS_REL_RPATH  =
 QMAKE_REL_RPATH_BASE    = @loader_path

 QMAKE_LIBS_DYNLOAD      =
-QMAKE_LIBS_OPENGL       = -framework OpenGL -framework AGL
+QMAKE_LIBS_OPENGL       = -framework OpenGL
 QMAKE_LIBS_THREAD       =

 QMAKE_INCDIR_WAYLAND    =
